doc/SAI_generate_debug_dump/SAI_generate_debug_dump.md

```diff
@@ -0,0 +1,262 @@
+    - [Terminology](#terminology)
+    - [Overview](#overview)
+    - [Requirements](#requirements)
+    - [Architecture Design](#architecture-design)
+    - [Implementation](#implementation)
+	  - [generate_sai_dump bash script](#generate_sai_dump-bash-script)
+	  - [show techsupport](#show-techsupport)
+	  - [DbgGenDump orchestration](#DbgGenDump-orchestration)
+	  - [SAI global API sai_dbg_generate_dump](#SAI-global-API-sai-dbg_generate_dump)
+	  - [syncd extended operation](#syncd-extended-operation)
+    - [YANG model changes](#yang-model-changes)
+    - [CLI](#cli)
+    - [Warmboot and Fastboot Design Impact](#Warmboot-and-Fastboot-Design-Impac)
+    - [Testing Requirements/Design](#Testing-Requirements/Design)
+    - [Unit Test cases](#Unit-Test-cases)
+    - [System Test cases](#System-Test-cases)
+
+### Revision
+
+| Rev |   Date   |          Author           | Change Description |
+| :-: | :------: | :-----------------------: | ------------------ |
+| 0.1 | 10/15/24 | Aviram Dali (**Marvell**) | Initial Draft      |
+### Scope
+
+The scope of this document is to design the handling of generating a SAI debug dump file by  user command , specifically for `show techsupport` command.
+
+### Terminology
+
+| Term  | Definition                              |
+| ----- | --------------------------------------- |
+| ASIC  | Application Specific Integrated Circuit |
+| SYNCD | ASIC Synchronization Service            |
+| SAI   | Switch Abstraction Interface            |
+| API   | Application Programmable Interface      |
+| SWSS  | Switch State Service                    |
+### Overview
+SAI dump file usually includes SDK info and configuration , SAI stats, capture of SAI lower layer states like registers vales etc...
+
+Currently, the SAI dump file is generated only during SAI failures by executing a dedicated executable named "saisdkdump" (which linkage with the SAI lib and during initialization it creates a new switch in redundant mode) 
+
+This new feature allows users to generate a SAI debug dump file using `show tech-support` command, not necessarily during failure.
+
+### Requirements
+
++ Add infrastructure to generate a SAI debug dump file upon user request
++ generate a SAI debug dump file from 'show techsupport' command.
++ Generate a SAI debug dump file within the context of Syncd.
++ Maintain the existing mechanism for generating the SAI debug dump file on failure.
+
+### Assumptions
+
+The SAI global API `sai_dbg_generate_dump` operates as a blocking function.
+
+### Architecture Design
+
+Adding a new infrastructure without changes in existing Sonic Architecture
+
+1. A user command, such as `show techsupport` triggers the `generate_sai_dump`, and creates a new table with the dump file name to create in the APPL DB.
+2. A new orchestration agent, `DbgGenDumpOrch`, is triggered to handle the request.
+3. `DbgGenDumpOrch` writes the file name to the ASIC DB and sets a new operation `REDIS_ASIC_STATE_COMMAND_DBG_GEN_DUMP` for syncd.
```

Comment:
This may be a bottle neck in the regular orchagent to syncd flow using ASIC_DB. As the commands are synchronous, during techsupport it may block other events.
If the saisdkdump takes time this may even cause timeout between orchagent and syncd. If the motivation is to just move from saisdkdump I don't see any advantage in this flow and it might event cause performance issues.

Reply:
This was discussed and consensus is to provide an ability to make the call non blocking. This should be configurable. @aviramd Please add this to the document